tools/gfx/d3d12/render-d3d12.cpp

```diff
@@ -3945,7 +3945,9 @@ class D3D12Device : public RendererBase
             for (uint32_t i = 0; i < fenceCount; ++i)
             {
                 auto fenceImpl = static_cast<FenceImpl*>(fences[i]);
-                m_d3dQueue->Wait(fenceImpl->m_fence.get(), waitValues[i]);
```

Comment:
I think? what you had here before was correct. 

That you want to move the 
```
waitValues[i] == kTimeoutInfinite ? INFINITE : waitValues[i] / 1000000);
```

Logic down to D3D12Device::waitForFences.

I suppose there is some confusion because they both have the same method name. Perhaps you want to rename this function to waitForFenceValues?

Reply:
Good catch, this is the wrong place.